datafusion/src/scalar.rs

```diff
@@ -283,6 +283,155 @@ impl ScalarValue {
         self.to_array_of_size(1)
     }
 
+    /// Converts an iterator of references [`ScalarValue`] into an [`ArrayRef`]
```

Comment:
I couldn't find any other way to take a bunch of `ScalarValues` and turn them back into an Array. Perhaps I missed something